src/mmw/js/src/modeling/models.js

```diff
@@ -517,10 +517,13 @@ var ScenariosCollection = Backbone.Collection.extend({
         });
 
         if (match) {
+            window.alert("There is another scenario with the same name. " +
+                    "Please choose a unique name for this scenario."
+            );
             console.log('This name is already in use.');
-            return;
+            return false;
         } else if (model.get('name') !== newName) {
-            model.set('name', newName);
+            return model.set('name', newName);
```

Comment:
This should hopefully take care of the unlikely edge case when a `model.set` fails (for example, if there are [validation errors](http://backbonejs.org/docs/backbone.html#section-68))